main/router.c

```diff
@@ -78,6 +78,8 @@ static void splitAndSend(const CPXRoutablePacket_t* rxp, RouteContext_t* context
     if (toSend > mtu) {
       toSend = mtu;
       lastPacket = false;
+    } else {
+      lastPacket = true;
```

Comment:
I don't think lastPacket should be set to true here, instead it should be set based on the incoming packet (as it used to be). The reason is that the incoming packet may have been split in an earlier router and setting it explicitly to true here may make it hard for the receiving application to understand how to assemble data. Only passing the incoming value on will make sure the end receiver only gets one lastPacket==true per transmitted packet (from the original transmitter)

Reply:
Yes, after discussing this there's a new commit for this.

Reply:
And yes, the discovery stuff was unintentional. I branched off from the wrong branch.